src/lib/datepicker/datepicker.spec.ts

```diff
@@ -1489,6 +1502,27 @@ describe('MatDatepicker', () => {
     }));
 
   });
+
+  describe('datepicker with custom header', () => {
+    let fixture: ComponentFixture<DatepickerWithCustomHeader>;
+    let testComponent: DatepickerWithCustomHeader;
+
+    beforeEach(fakeAsync(() => {
+      fixture = createComponent(
+        DatepickerWithCustomHeader,
+        [MatNativeDateModule],
+        [],
+        [CustomHeaderForDatepicker],
+        [CustomHeaderForDatepicker]
+      );
+      fixture.detectChanges();
+      testComponent = fixture.componentInstance;
+    }));
+
+    it('should instantiate a datepicker with a custom header', fakeAsync(() => {
+      expect(testComponent).toBeTruthy();
```

Comment:
One more thing to check:

```ts
expect(fixture.debugElement.nativeElement.querySelector('mat-calendar-header')).toBeTruthy();
```

Reply:
Hm, it does not find it.

Do I need to do `testComponent.datepicker.open();` before?

Reply:
I imagined a thing like this:

```
it('should find the standard header selector', fakeAsync(() => {
      testComponent.datepicker.open();
      fixture.detectChanges();
      expect(fixture.debugElement.nativeElement.querySelector('mat-calendar-header')).toBeTruthy();
    }));
```

Reply:
Did you try `fixture.nativeElement.querySelector(...)` (with no `debugElement`)? I see this elsewhere in the file too. Using `document.querySelector` doesn't feel quite right...

Reply:
I know what you mean, you want to get the content from the parent element. 

Unfortunately `fixture.nativeElement.querySelector('mat-calendar-header')` yields `null`.

Reply:
When I do `console.log(fixture.nativeElement);` I get

```html
<div id="root1" ng-version="6.0.0-rc.5"><input ng-reflect-mat-datepicker="[object Object]" aria-haspopup="true" aria-owns="mat-datepicker-1"><mat-datepicker></mat-datepicker></div>
```

This corresponds with the selectors in the template of `DatepickerWithCustomHeader`. The selectors should be replaced by their associates templates so we could query for the actual DOM-elements.

Reply:
Oh I think I understand what's happening. The overlay must be adding itself to `document.body` which is outside the fixture. Ok `document.querySelector` is fine then

Reply:
@mmalerba In case there are more things left to do for this PR, please let me know. I can take care of it tomorrow or Friday. Then I will be away for two weeks.